components/src/lib/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,295 @@
+import { Meta, StoryObj } from '@storybook/react';
+import { Modal } from './Modal';
+import { useState } from 'react';
+import Button from '../Button/Button';
+import useModalContext from './hooks/useModalContext';
+import Input from '../Input/Input';
+
+const meta: Meta<typeof Modal> = {
+  title: 'Components/Modal',
+  component: Modal,
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+};
+
+export default meta;
+type Story = StoryObj<typeof Modal>;
+
+export const Basic: Story = {
+  render: () => (
+    <Modal>
+      <Modal.Trigger>
+        <Button>기본 모달 열기</Button>
+      </Modal.Trigger>
+      <Modal.Content>
+        <Modal.Header>
+          <div>
+            <Modal.Title>기본 모달</Modal.Title>
+            <Modal.Description>기본 모달 컴포넌트 설명</Modal.Description>
+          </div>
+          <Modal.CloseButton />
+        </Modal.Header>
+        <div>
+          <p>기본 모달 내용</p>
+        </div>
+      </Modal.Content>
+    </Modal>
```

Comment:
상태를 직접 정의해주긴 싫지만, 그 상태를 활용하고 싶을 수도 있는데요,
그럴 땐 아래처럼 renderProps를 활용하면 유용합니다!

```suggestion
    <Modal>
      {({ isOpen }) => 
        <>
          <Modal.Trigger>
            <Button>{isOpen ? '이미 열림' : '기본 모달 열기'}</Button>
          </Modal.Trigger>
          <Modal.Content>
            <Modal.Header>
              <div>
                <Modal.Title>기본 모달</Modal.Title>
                <Modal.Description>기본 모달 컴포넌트 설명</Modal.Description>
              </div>
              <Modal.CloseButton />
            </Modal.Header>
            <div>
              <p>기본 모달 내용</p>
            </div>
          </Modal.Content>
        </>
    </Modal>
```

Reply:
이런 방식으로도 구현할 수 있네요 ,,
쵸파의 리뷰 덕분에 `렌더 프롭 패턴`에 대해서 알 수 있었습니다. 👍👍

렌더 프롭 패턴은 컴포넌트 내부에 있는 `isOpen`이라는 상태를 사용자가 사용하고 싶을 때 사용되는군요! 직접 써보지는 않았지만, 모달의 열림/닫힘 상태에 따라 사용자가 조건부 렌더링을 한다던지, 다른 스타일을 적용시키고 싶을 때 유용할 것 같아요.